study-group/Toast-Implementation-Studies.md

```diff
@@ -70,25 +70,164 @@ The `<paper-toast>` holds onto information about how it was dismissed the last t
+  normalToast = () => toast("This is a normal toast");
+
+  render(){
+    return (
+      <div>
+        <button onClick={this.tenSecondToast}>Close after 10 seconds</button>
+        <button onClick={this.normalToast}>Close after default amount of time</button>
+        <ToastContainer autoClose={5000} />
+      </div>
+    );
+  }
+}
+```
+
+The first button creates a toast that dismisses in 10 seconds, overriding the `ToastContainer` default.
+The second provides no `autoClose` value, so it dismisses in the default 5 seconds.
+For some of the more common controls, the toast can be configured like so:
+
+```js
+class App extends Component {
+  topRightWarning = () => toast('Warning!', {
+    type: toast.TYPE.WARNING,
+    position: toast.POSITION.TOP_RIGHT,
+    onClose: ({ props }) => window.alert('I warned you') // props: all props passed to rendered toast component
+  });
+
+  render() {
+    return (
+      <button onClick={this.topRightWarning}>Warn me</button>
+      <ToastContainer autoClose={5000} />
+    );
+  }
+}
+```
+
+`react-toastify` supports built-in themes for toasts, such as `success`, `info`, and `warning`, among others.
+It also provides intuitive positioning, that can be set from the toast or configured on the `ToastContainer`.
+Additionally, callbacks like `onClose` and `onOpen` allow the toast to have intelligent effects when the user interacts with it.
+
+### Notable Features & Details
+
+#### Toast ID
+the `toast()` function returns an ID of the toast it creates, which can be used to act upon that toast.
+For instance, with the toast ID the toast can be dismissed, updated, and checked to see if it's active.
+
+#### Mobile
+On mobile toasts will automatically take all available width, and all toasts are swipeable / draggable to dismiss.
+
+#### Styling
+Library provides fields `className`, `toastClassName`, and, `progressClassName` for the developer to pass in CSS classes.
+
+#### Render Custom Component
+To customize the view the `toast` function can take in a React component, which will receive the function required to close the toast as props (though it won't get the toast ID).
+
+#### cssTransitionHelper
+The library provides a `cssTransitionHelper` helper function to customize animations without having to work directly with `react-transition-group`.
+The helper allows the dev to set classes and durations for `enter` and `exit`, as well as make transitions position specific.
+
+### Takeaways
+- Extremely unopinionated and configurable; the developer has near total control.
+- Component pattern is optional, only needs one component, but allows multiple.
+- Close button is default, but can be replaced with a custom close button.
+  - Separate from passing in a custom React Component.
+- The toast can be configured to display after a certain delay.
+  - I'm not sure what the advantage this has over firing the toast in a setTimeout.
+- Passing null into a property resets it to default (useful for `update()`).
```

Comment:
What is update()? It's not referenced above.

Reply:
Was not clear enough with this, changing it now